src/lib/ccf/index.ts

```diff
@@ -0,0 +1,458 @@
+      let architecture = instance['Microarchitecture'];
+      if (!(architecture in this.azureList)) {
+        architecture = 'Average';
+      }
+      this.computeInstances['azure'][instance['Virtual Machine']] = {
+        consumption: {
+          minWatts: this.azureList[architecture]['Min Watts'] * cpus,
+          maxWatts: this.azureList[architecture]['Max Watts'] * cpus,
+        },
+        name: instance['Virtual Machine'],
+        vCPUs: instance['Instance vCPUs'],
+        maxvCPUs: parseInt(
+          instance['Platform vCPUs (highest vCPU possible)'],
+          10
+        ),
+      } as IComputeInstance;
+    });
+    aws_embodied.forEach((instance: KeyValuePair) => {
+      this.computeInstances['aws'][instance['type']].embodiedEmission =
+        instance['total'];
+    });
+    gcp_embodied.forEach((instance: KeyValuePair) => {
+      this.computeInstances['gcp'][instance['type']].embodiedEmission =
+        instance['total'];
+    });
+    azure_embodied.forEach((instance: KeyValuePair) => {
+      this.computeInstances['azure'][instance['type']].embodiedEmission =
+        instance['total'];
+    });
+  }
+
+  // Architecture strings are different between Instances-Use.JSON and the bundled Typescript from CCF.
+  // This function resolves the differences.
+  private resolveAwsArchitecture(architecture: string) {
+    if (architecture.includes('AMD ')) {
+      architecture = architecture.substring(4);
+    }
+
+    if (architecture.includes('Skylake')) {
+      architecture = 'Sky Lake';
+    }
+
+    if (architecture.includes('Graviton')) {
+      if (architecture.includes('2')) {
+        architecture = 'Graviton2';
+      } else if (architecture.includes('3')) {
+        architecture = 'Graviton3';
+      } else {
+        architecture = 'Graviton';
+      }
+    }
+
+    if (architecture.includes('Unknown')) {
+      architecture = 'Average';
+    }
+
+    if (!(architecture in this.awsList)) {
+      console.log('ARCHITECTURE:', architecture);
+    }
+
+    return architecture;
+  }
+
+  /**
+   * Calculates the embodied emissions for a given observation
+   */
+  private embodiedEmissions(observation: KeyValuePair): number {
+    // duration
+    const duration_in_hours = observation['duration'] / 3600;
+    // M = TE * (TR/EL) * (RR/TR)
+    // Where:
+    // TE = Total Embodied Emissions, the sum of Life Cycle Assessment(LCA) emissions for all hardware components
+    // TR = Time Reserved, the length of time the hardware is reserved for use by the software
+    // EL = Expected Lifespan, the anticipated time that the equipment will be installed
+    // RR = Resources Reserved, the number of resources reserved for use by the software.
+    // TR = Total Resources, the total number of resources available.
+    const TotalEmissions =
+      this.computeInstances[this.provider][this.instanceType]
+        .embodiedEmission ?? 0;
+    const TimeReserved = duration_in_hours;
```

Comment:
let's declare variables starting with lowercase since Uppercase first letter is associated with Classes